APKBUILD.5

```diff
@@ -0,0 +1,488 @@
+
+.It Cm !check
+Specifies that the package will not run a test suite.  The reason for disabling
+the check phase should be noted in a comment.
+
+.It Cm !checkroot
+Specifies that the package's test suite will be run as a non-privileged user
+instead of using
+.Xr fakeroot 8 .
+This is necessary for some test suites which fail when run as root.
+
+.It Cm !dbg
+(Adélie-specific) Specifies that the package should not be built with a debug
+information package.  This is typically used on packages that do not generate
+debug information (such as pure Python packages) or packages that do not
+support debug information packages (such as LLVM).
+
+.It Cm !fhs
+Specifies that the package violates FHS and installs to a location such as
+.Pa /usr/local ,
+.Pa /opt ,
+or
+.Pa /srv .
+
+.It Cm ldpath-recursive
+Specifies that
+.Xr abuild 1
+should use the
+.Fl --recursive
+argument to
+.Xr scanelf 1
+when attempting to find shared library (.so) dependencies for the package.
+
+.It Cm !libc_eglibc
+Specifies that the package cannot be built on the eglibc libc.
+
+.It Cm !libc_musl
+Specifies that the package cannot be built on the musl libc.
+
+.It Cm !libc_uclibc
+Specifies that the package cannot be built on the uclibc libc.
+
+.It Cm libtool
+Specifies that the package requires its libtool (.la) files.  They will not be
+automatically removed by
+.Xr abuild 1 .
+
+.It Cm net
+Specifies that the package build system requires access to a network.  This is
+discouraged and an issue should be filed with the package's authors.
+
+.It Cm !strip
+Specifies that
+.Xr strip 1
+should not be run on any of the package's binaries.  This is automatically
+implied if the -dbg subpackage is enabled, or if you are using
+.Ev DEFAULT_DBG .
+
+.It Cm suid
+Specifies that binaries in the package may be installed set-uid.  This is a
+security risk and it is highly recommended to use capabilities or process
+separation instead of set-uid where available.
+
+.It Cm textrels
+Specifies that the package's binaries 
+
+.It Cm toolchain
+Specifies that the package is part of the base toolchain set and may depend
+on packages like
+.Li g++ .
+
+.It Cm !tracedeps
+Specifies that
+.Xr abuild 1
+should not automatically populate
+.Cm depends
+with shared library (.so) or symlink target dependencies.
+
+.El
+
+
+.Ss Special Variables
+The following variables are used only in special circumstances, and may be
+required or optional depending on their usage and the contents of other
+variables.
+
+.Bl -tag -width Ds
+.It Cm depends_dev
+Specifies the run-time dependencies of the -dev subpackage.
+
+.It Cm giturl
+Specifies the URL of the Git repository to use with
+.Cm abuild checkout .
+If the default branch of the repository is not desired, a different one may be
+specified by appending
+.Fl b Ar branch
+where
+.Cm branch
+is the branch to checkout.
+.El
+
+
+.Ss Functions
+Functions specified here may be present in any
+.Nm
+file, but with the exception of
+.Li build and
```

Comment:
build function is not required (it used to be).

Reply:
It looks like [`package` is optional](https://github.com/alpinelinux/abuild/blob/master/abuild.in#L136) as well.  So I guess all functions are optional?

Reply:
No, package is not optional, because when you omit package, `pkgdir` is not created and abuild fails with error that `pkgdir` does not exist. This is a safety check.